src/cloudai/configurator/env_params.py

```
@@ -0,0 +1,212 @@
+    """
+
+    model_config = ConfigDict(extra="forbid")
+
+    kind: Literal["box", "discrete"]
+    dim: int = 1
+    n: Optional[int] = None
+
+    @model_validator(mode="after")
+    def _validate(self) -> Self:
+        if self.dim < 1:
+            raise ValueError(f"ObsLeafDescriptor dim must be >= 1; got {self.dim}")
```

Ruff (0.15.17)

[warning] Avoid specifying long messages outside the exception class

(TRY003)